api/envoy/config/core/v3/health_check.proto

```diff
@@ -163,6 +170,10 @@ message HealthCheck {
     // When checking the response, “fuzzy” matching is performed such that each
     // binary block must be found, and in the order specified, but not
     // necessarily contiguous.
+    //
+    // .. note::
+    //
+    //   The text payload should be Hex encoded payload. E.g., "000000FF", and binary is not implemented.
```

Comment:
Why aren't thee constraints left on `Payload`? Aren't they true wherever `Payload` is used?

Reply:
The health check for TCP is using repeated Hex string Payload which will be decoded to bytes. While for HTTP payload, users can specify one UTF-8 string payload as an expected response body for convenient purpose, or they can set the binary to follow the HTTP standard. The UTF-8 string is using by Google right-now.

Reply:
Can we be consistent here and support the full range of options? It seems this is just a protobuf config utility thing and easy to support the same formats across both if the matcher is normalized internally?

Reply:
Good point. Yes, extended the Payload, and will not impact the current TCP health check usage.

Reply:
Emm.. looks like an enum is better, changed it. Thank you!

Reply:
Looks fine to me now I think, thanks. I think if we did it from scratch we would use some standard generic matchers but given the Payload prior art in HCs this is the right way to go. Over to you @adisuissa.